tf2onnx/graph_matcher.py

```diff
@@ -102,21 +102,33 @@ def add(self, pattern, op, tensor):
             #       'Name %s is already bound to another pattern' % pattern.name)
             self._name_to_pattern[pattern.name] = pattern
 
-    def _to_pattern(self, pattern_or_name):
+    def _to_pattern(self, pattern_or_name, default=None):
+        """
+        if pattern_or_name cannot match OptypePattern and is not in name_to_pattern, then the function return None
+        """
         if isinstance(pattern_or_name, OpTypePattern):
             return pattern_or_name
 
         if isinstance(pattern_or_name, six.text_type):
-            return self._name_to_pattern[pattern_or_name]
+            get_name_to_pattern = self._name_to_pattern.get(pattern_or_name)
+            if get_name_to_pattern is None:
+                return default
+            return get_name_to_pattern
 
         raise ValueError('pattern_or_name has type %s. Expect OpTypePattern or str.'
                          % type(pattern_or_name))
 
-    def get_op(self, pattern_or_name):
-        return self._pattern_to_op_tensor[self._to_pattern(pattern_or_name)][0]
-
-    def get_tensor(self, pattern_or_name):
-        return self._pattern_to_op_tensor[self._to_pattern(pattern_or_name)][1]
+    def get_op(self, pattern_or_name, default=None):
+        pattern = self._to_pattern(pattern_or_name)
+        if not pattern:
+            return default
+        return self._pattern_to_op_tensor[pattern][0]
```

Comment:
I still prefer using `get` to fetch op/tensor: 
```
op_and_tensor = self._pattern_to_op_tensor.get(self._to_pattern(pattern_or_name))
if op_and_tensor:
    return op_and_tensor[0]
else:
    return default
```

Reply:
Thanks. I have improved the codes in the functions and  supplemented the comments.